### PR TITLE
Vectorized form of the for loop

### PR DESCRIPTION
**Pull request by PragyanTiwari**
Replaced for loop with its vectorized form using numpy for increasing efficiency while dealing with large data in the src file : `sentiments.py`

```
import pandas as pd
import numpy as np

# dropping na values 
transaction_shares = pd.Series([t['trade'] for t in insider_trades]).dropna()

# vectorized form of the previous loop for more efficiency while dealing with large data.
bearish_condition = transaction_shares < 0
signals = np.where(bearish_condition,
                x="bearish", # if condition is True
                y="bullish").tolist() # if condition is False
```

previous src code:

```
signals = []
for trade in insider_trades:
    transaction_shares = trade["transaction_shares"]
    if not transaction_shares:
       continue
    if transaction_shares < 0:
       signals.append("bearish")
    else:
        signals.append("bullish")

### CODE REVIEW

**Comment by PragyanTiwari** [comment]
Hey @virattt , 
Could you give me an update on this pull request?
I know not to tag the maintainers, but I wanted to know if you had thought about the scr file update.

File: src/agents/sentiment.py [hunk 1]
```diff
@@ -13,15 +17,15 @@ def sentiment_agent(state: AgentState):
...
-        transaction_shares = trade["transaction_shares"]
-        if not transaction_shares:
-            continue
-        if transaction_shares < 0:
-            signals.append("bearish")
-        else:
-            signals.append("bullish")
+
+    # dropping na values 
+    transaction_shares = pd.Series([t['trade'] for t in insider_trades]).dropna()
```
Comment (virattt) [inline src/agents/sentiment.py:22]: Hey @PragyanTiwari - "trade" is not a valid key in the `t` here.

Perhaps you meant `"transaction_shares"`?

Comment (virattt) [inline src/agents/sentiment.py:22]:
```suggestion
    transaction_shares = pd.Series([t['transaction_shares'] for t in insider_trades]).dropna()
```

File: src/agents/sentiment.py [hunk 2]
```diff
@@ -13,15 +17,15 @@ def sentiment_agent(state: AgentState):
...
+    
+    # vectorized form of the previous loop for more efficiency while dealing with large data.
+    bearish_condition = transaction_shares < 0
+    signals = np.where(bearish_condition,
+                       x="bearish", # if condition is True
+                       y="bullish").tolist() # if condition is False
```
Comment (virattt) [inline src/agents/sentiment.py:28]: can be simplified to:

```suggestion
    signals = np.where(bearish_condition, "bearish", "bullish").tolist()
```

**Comment by virattt** [comment]
Thanks!

**Pull request merged**